src/P5Functions/Math.js

```diff
@@ -433,7 +433,14 @@ module.exports = function withMath(p, undef) {
       return internalRandomGenerator() * arguments[0];
     }
     var aMin = arguments[0], aMax = arguments[1];
-    return internalRandomGenerator() * (aMax - aMin) + aMin;
+    while (true) {
+      var ir = internalRandomGenerator();
+      var result = ir * (aMax - aMin) + aMin;
+      if (result !== aMax || result === aMin) {
```

Comment:
Though it may be a programming error, the degenerate invocation of `random(x, x)` always returns `x`.  My original code guaranteed that the loop was not infinite and that the degenerate behavior remained the same.  The first expression of the OR tested for the desired  (and expected) case.  The second clause tested for the degenerate case.

The proposed edit is incorrect when the second argument is less than the first, e.g. `random(10, -5);`   Also note that swapping `random(b, a);` for `random(a, b);` does not maintain the existing functionality or the functionality described by the function's header's comment.